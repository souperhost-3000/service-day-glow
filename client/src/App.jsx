import React from 'react';

class App extends React.Component {
  render() {
    return (
      <div>This is a React rendered page</div>
    )
  }
}

export default App;