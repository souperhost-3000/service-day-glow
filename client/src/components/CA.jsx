/* eslint-disable no-alert */
import React from 'react';

// check availability button
function CA({ setCostEst, callToAction }) {
  return (
    <div className="CA-app">
      <button
        className="CA-btn"
        type="button"
        onClick={() => {
          if (callToAction === 'Check availability') {
            setCostEst(true);
          } else {
            alert('Please sign in to complete your reservation');
          }
        }}
      >
        {callToAction}
      </button>
    </div>
  );
}

export default CA;
