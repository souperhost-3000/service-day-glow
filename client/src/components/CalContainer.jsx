/* eslint-disable array-callback-return */
/* eslint-disable consistent-return */
/* eslint-disable react/prop-types */
import React, { useState, useEffect } from 'react';
import Calendar from './Modals/Calendar';

// Calendar Container (reuseable) - formats each month of availability for listing
function CalContainer({
  availability, subHeader, adjPrice, updateDate, checkIn, checkOut,
}) {
  const [leftCalIdx, setLeftCal] = useState(0);
  const [rightCalIdx, setRightCal] = useState(leftCalIdx + 1);
  const [pricePerNight, setPricePerNight] = useState(0);

  useEffect(() => {
    setRightCal(leftCalIdx + 1);
  }, [leftCalIdx]);

  useEffect(() => {
    setPricePerNight(adjPrice);
  }, [adjPrice]);

  function selectedDates(e, check) {
    updateDate(e, check);
  }

  return (
    <div className="months-container" key="leftCalIdx">
      {availability.map((monthObj, index) => {
        // only 2 selected months for the display (not every month in container)
        if (index === leftCalIdx || index === rightCalIdx) {
          return (
            <div className={index === leftCalIdx ? 'cal-left' : 'cal-right'}>
              <Calendar
                monthName={monthObj.name}
                key={monthObj.name}
                index={index}
                monthDays={monthObj.days}
                side={index === leftCalIdx ? 'cal-left' : 'cal-right'}
                setLeftCal={setLeftCal}
                pricePerNight={pricePerNight}
                subHeader={subHeader}
                adjPrice={adjPrice}
                selectedDates={selectedDates}
                mainCheckIn={checkIn}
                mainCheckOut={checkOut}
              />
            </div>
          );
        }
      })}
    </div>
  );
}

export default CalContainer;
