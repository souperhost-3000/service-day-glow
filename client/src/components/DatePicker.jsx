/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { useState, useEffect } from 'react';
import CalModal from './Modals/CalModal';
import DateBoxes from './DateBoxes';

// date selectors (check-in, check-out)
function DatePicker({
  availability, updateDate, subHeader, adjPrice, showCalModal, checkIn, checkOut,
}) {
  const [showModal, setModal] = useState(false);

  useEffect(() => {
    setModal(showCalModal);
  }, [showCalModal]);

  return (
    <div>
      <div className="datePicker" onClick={() => setModal(true)}>
        <DateBoxes
          updateDate={updateDate}
          checkIn={checkIn}
          checkOut={checkOut}
        />
      </div>
      <div className="modal-middle">
        <span>
          <CalModal
            show={showModal}
            setCalModal={setModal}
            availability={availability}
            subHeader={subHeader}
            adjPrice={adjPrice}
            updateDate={updateDate}
            checkIn={checkIn}
            checkOut={checkOut}
          />
        </span>
      </div>
    </div>
  );
}

export default DatePicker;
