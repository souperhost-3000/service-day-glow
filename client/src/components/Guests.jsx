import React, { useState, useEffect } from 'react';
import GuestModal from './Modals/GuestModal';

// Guest selector (num adults, children, infants)
function Guests({ guestLimit, updateGuestTotal, guestTotal }) {
  const [showGMod, setGuestModal] = useState(false);
  // const [guestTotal, setGuestTotal] = useState(1);

  useEffect(() => {
    if (showGMod) {
      document.getElementById('guest-expand-mod').style.transform = 'rotate(-180deg)';
    }
    if (!showGMod) {
      document.getElementById('guest-expand-mod').style.transform = 'rotate(0deg)';
    }
  }, [showGMod]);

  return (
    <div className="guestPicker">
      <div>
        <span id="guest-title">Guests</span>
        <br />
        <span id="guestCount">
          {`${guestTotal} guest${guestTotal > 1 ? 's' : ''}`}
        </span>
        <button id="guest-expand-mod" type="button" onClick={() => setGuestModal(!showGMod)}>
          {/* ╲╱ */}
        </button>
      </div>
      {/* <button id="guest-collapse-mod"> ╱╲</button> */}
      <div className="guestPicker-modal">
        <span>
          <GuestModal
            showGMod={showGMod}
            setGuestModal={setGuestModal}
            updateGuestTotal={updateGuestTotal}
            guestLimit={guestLimit}
          />
        </span>
      </div>
    </div>
  );
}

export default Guests;
