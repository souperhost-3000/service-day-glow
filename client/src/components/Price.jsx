import React from 'react';

// dynamic price goes here
function Price({ adjPrice }) {
  return (
    <div className="price-app">
      <span id="price-app-header">
        {` $${adjPrice} `}
      </span>
      <span id="suffix">/ night</span>
    </div>
  );
}

export default Price;
