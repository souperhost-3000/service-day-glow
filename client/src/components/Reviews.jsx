/* eslint-disable camelcase */

import React from 'react';

// dynamic reviews goes here
function Reviews({ rating, reviews_count }) {
  return (
    <div className="reviews">
      <img id="star-sym" src="https://ghrsea12-fec.s3-us-west-2.amazonaws.com/sample/star.png" alt="" />
      <span className="rating-scored" id="stars">
        {`  ${rating}`}
      </span>
      <span className="reviewsCount" id="numReviews">
        {`  (${reviews_count})`}
      </span>
    </div>
  );
}

export default Reviews;
