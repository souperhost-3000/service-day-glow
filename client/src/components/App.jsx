/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import '../styles/app.css';
import Price from './Price';
import Reviews from './Reviews';
import DatePicker from './DatePicker';
import CA from './CA';
import Guests from './Guests';
import exampleData from '../tests/exampleData';
import CalContainer from './CalContainer';
import CostEstModal from './Modals/CostEstModal';

// top level service component (displayed before user interacts with anything)
function App() {
  const [showCostEst, setCostEst] = useState(false);
  const [style, setStyle] = useState({ height: '260px' });
  const [callToAction, setCallToAction] = useState('Check availability');
  const [subHeader, setSubHeader] = useState('');
  const [adjPrice, setAdjPrice] = useState(0);
  const [showCalModal, setCalModal] = useState(false);
  const [listingData, setListingData] = useState(exampleData);
  const listingID = 11;

  // GET request for Listing Data
  useEffect(() => {
    axios.get(`/availability/${listingID}`)
      .then((response) => setListingData(response.data))
      // eslint-disable-next-line no-console
      .catch((err) => console.log(err));
  }, [listingID]);

  const [checkOut, setCheckOut] = useState('');
  const [checkIn, setCheckIn] = useState('');
  const [numNights, setNumNights] = useState(1);

  function updateDate(e, check) {
    if (check === 'in') {
      setCheckIn(e);
    }
    if (check === 'out') {
      setCheckOut(e);
    }
  }

  // calculate number of nights for selected dates
  useEffect(() => {
    if (checkIn && checkOut) {
      if (checkIn.slice(0, 1) === checkOut.slice(0, 1)) {
        const nights = Number(checkOut.slice(3, 5)) - Number(checkIn.slice(3, 5));
        setNumNights(nights);
      }
    }
  }, [checkIn, checkOut]);

  // changes subHeader details based on user's interaction (&#8226;)
  useEffect(() => {
    if (checkIn && checkOut) {
      setSubHeader(`${checkIn} - ${checkOut}`);
    } else if (checkIn) {
      setSubHeader('Minimum stay: 2 nights');
    } else if (!checkIn && !checkOut) {
      setSubHeader(`${listingData.listing_type} . ${listingData.beds} beds . ${listingData.baths} bath`);
    }
  }, [listingData, checkIn, checkOut]);

  // const [guestDetails, setGuestDetails] = useState(1);
  const [guestTotal, setGuestTotal] = useState(1);
  function updateGuestTotal(e) {
    setGuestTotal(guestTotal + e);
  }

  // when CA button is clicked, expands app major & changes text on CA button
  useEffect(() => {
    if (showCostEst) {
      setStyle({ height: '500px' });
      setCallToAction('Reserve now');
    }
  }, [showCostEst]);

  // as guest total changes, adjust nightly rate
  useEffect(() => {
    if (guestTotal > listingData.guest_included) {
      const extraGuestCount = (guestTotal - listingData.guest_included);
      setAdjPrice((listingData.guest_extra_charge * extraGuestCount) + listingData.price);
    } else {
      setAdjPrice(listingData.price);
    }
  }, [guestTotal, listingData]);

  return (
    <div className="entire-App" onClick={() => setCalModal(false)}>
      <div className="scroll-wrapper">
        <div className="header-container">
          <div className="header">
            <div className="description-title">
              {`${listingData.listing_type} hosted by ${listingData.host}`}
            </div>
            <h3>
              {`${listingData.guest_limit} guests . ${listingData.bedrooms} bedrooms . ${listingData.beds} beds . ${listingData.baths} bath`}
            </h3>
          </div>
          <div className="listing-static-info">
            <div>
              <img id="listing-static-top" src="https://ghrsea12-fec.s3-us-west-2.amazonaws.com/sample/airbnbStatic1.webp" alt="" />
            </div>
            <div>
              <img id="listing-static-bottom" src="https://ghrsea12-fec.s3-us-west-2.amazonaws.com/sample/airbnbStatic2.webp" alt="" />
            </div>
          </div>
        </div>
        <section className="major">
          <div className="major-container">
            <div className="app-container" style={style}>
              <div className="upper-app">
                <Price adjPrice={adjPrice} />
                <Reviews
                  rating={listingData.rating}
                  reviews_count={listingData.reviews_count}
                />
              </div>
              <div className="middle-app">
                <DatePicker
                  availability={listingData.availability}
                  subHeader={subHeader}
                  adjPrice={adjPrice}
                  showCalModal={showCalModal}
                  updateDate={updateDate}
                  checkIn={checkIn}
                  checkOut={checkOut}
                />
                <Guests
                  guestLimit={listingData.guest_limit}
                  updateGuestTotal={updateGuestTotal}
                  guestTotal={guestTotal}
                />
              </div>
              <div className="lower-app">
                <CA
                  setCostEst={setCostEst}
                  callToAction={callToAction}
                />
                <CostEstModal
                  listingData={listingData}
                  showCostEst={showCostEst}
                  adjPrice={adjPrice}
                  numNights={numNights}
                />
              </div>
            </div>
          </div>
        </section>
        <section className="minor">
          <div className="minor-container">
            <div className="minor-header">
              <div className="modal-header-left">
                <div className="selectDates">
                  Select Dates
                </div>
                <span className="listing_rooms">{subHeader}</span>
              </div>
            </div>
            <div className="cal-dbl-container">
              <div className="dbl-all">
                <CalContainer
                  availability={listingData.availability}
                  subHeader={subHeader}
                  adjPrice={adjPrice}
                  updateDate={updateDate}
                  checkIn={checkIn}
                  checkOut={checkOut}
                />
              </div>
            </div>
            <div className="actions">
              <div className="taxesWarning">
                Prices on calendar do not include taxes and fees
              </div>
              <button
                className="clear-btn"
                type="button"
              >
                Clear date
              </button>
            </div>
          </div>
        </section>
      </div>
    </div>
  );
}

export default App;
