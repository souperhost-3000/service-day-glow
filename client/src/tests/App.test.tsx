import React from 'react';
import { shallow, mount, render } from 'enzyme';

import App from '../components/App';
import Price from '../components/Price';
import Reviews from '../components/Reviews';
import DatePicker from '../components/DatePicker';
import Guests from '../components/Guests';
import CA from '../components/CA';
// import Calendar from '../components/Modals/Calendar';
import CalContainer from '../components/CalContainer';

/*
=======================================
Unit Testing App Level - Overall Page
=======================================
*/

describe('Page basics at Main App level', () => {
  it('page renders App to service without breaking', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists()).toBe(true);
  });

  // it('App page contains Header to intro the service', () => {
  //   const wrapper = shallow(<App />);
  //   const header = wrapper.find("h2");
  //   const result = header.text();

  //   expect(result).toBe("AirbnbCheck AvailabilityService by Jacki");
  // });

  it('Minor - calendar has header', () => {
    const wrapper = shallow(<App />);
    const calHeader = wrapper.find("div.selectDates").text();

    expect(calHeader).toBe("Select Dates");
  })

});

/*
=======================================
Unit Testing App Level - Major
=======================================
*/

describe('Components nested in Major render', () => {

  it('should render Price and Reviews', () => {
    const wrapper = shallow(<App />);
    const price = wrapper.find(Price);
    const reviews = wrapper.find(Reviews);

    expect(price.exists()).toBe(true);
    expect(reviews.exists()).toBe(true);
  })

  it('should render DatePicker and GuestsPicker', () => {
    const wrapper = shallow(<App />);
    const datePicker = wrapper.find(DatePicker);
    const guests = wrapper.find(Guests);

    expect(datePicker.exists()).toBe(true);
    expect(guests.exists()).toBe(true);
  })

  it('should render CA button', () => {
    const wrapper = shallow(<App />);
    const caButton = wrapper.find(CA);

    expect(caButton.exists()).toBe(true);
  })

});

/*
=======================================
Unit Testing App Level - Minor
=======================================
*/

describe('Components nested in Minor render', () => {

  it('should render two Calendars', () => {
    const wrapper = shallow(<App />);
    const twoCal = wrapper.find(CalContainer);

    expect(twoCal.exists()).toBe(true);
  })

  it('should render clear button', () => {
    const wrapper = shallow(<App />);
    const clearBtn = wrapper.find("button.clear-btn");
    const clearBtntxt = clearBtn.text();

    expect(clearBtn.exists()).toBe(true);
    expect(clearBtntxt).toBe("Clear date");
  })

});

// app-level (each shallow test only tests one function)
// test GET req (data received)
// test price and review rendered (from GET req)
// calendar input is a form
// calendar expands modal
// guest input expands modal
// button rendered