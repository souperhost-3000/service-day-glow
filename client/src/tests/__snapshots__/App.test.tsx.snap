// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with enzyme and jest dynamically renders price 1`] = `ReactWrapper {}`;

exports[`Testing with enzyme and jest renders app service without breaking 1`] = `ReactWrapper {}`;
